### DISCUSSION TITLE
Binding redirects needs to be added for .Net Web projects in order to resolve the .Net assemblies correctly.

### DISCUSSION THREAD

**Discussion by sydseter** [renovatebot/renovate#24824] [Suggest an Idea]
### What would you like Renovate to be able to do?

Automatic binding redirects are implemented differently for .Net web apps. Because the source configuration (web.config) file must be modified for web apps, binding redirects are not automatically added to the configuration file. However, Visual Studio notifies you of binding conflicts, and you can add binding redirects to resolve the conflicts. Because you're always prompted to add binding redirects, you don't need to explicitly disable this feature for a web app. see: https://learn.microsoft.com/en-us/dotnet/framework/configure-apps/how-to-enable-and-disable-automatic-binding-redirection#enable-automatic-binding-redirects-in-web-apps

This means that it is not possible to use a system like renovate in .Net web projects because it only update the package references. In order for renovate to automaticly create pull-requests and merge them, the binding redirects needs to be updated. Too.

### If you have any ideas on how this should be implemented, please tell us here.

To solve this renovate probably needs a configuration value for enabling editing .Net assemblies in web.config so that it compares the binding redirects currently added to to the web.config with the incoming package reference update. 
There is a nuget package that adds the binding redirect, but the command needs to be run after the package reference has been added. https://blog.maartenballiauw.be/post/2014/11/27/could-not-load-file-or-assembly-nuget-assembly-redirects.html 
This could potentially solve the issue if renovate run `Get-Project -All | Add-BindingRedirect` after applying the new package reference.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by sydseter** [comment]
Question: Could this be used to run ´Get-Project –All | Add-BindingRedirect´ https://docs.renovatebot.com/configuration-options/#postupgradetasks
?

**Comment by rarkins** [comment]
That appears to be a GUI option, so if so then the answer is "No" - it would need to be a CLI capability

**Comment by viceice** [comment]
binding redirects are only needed on full dotnet framework, dotnet core simply uses the assembly found locally.

I'm not sure if it's worth implementing it into renovate.

**Comment by sydseter** [comment]
It's the most used framework worldwide. Whether this is a valid criteria depends on your business model I guess. https://www.statista.com/statistics/793840/worldwide-developer-survey-most-used-frameworks/

**Comment by sydseter** [comment]
Btw. for reference. There are solutions that others have tried to work out to solve this without requiring visual studio. All though they are immature, they do exist. https://andrewharcourt.com/articles/command-line-add-bindingredirect

**Comment by sydseter** [comment]
I found a way to solve this. I will need to test this out some more to see how this works, but I basically took the Add-Bindingredirect project from Andrew Harcourt and upgraded the dependencies, then I built a new release for both windows and linux. https://github.com/AdmincontrolAS/Add-BindingRedirect/releases

This allows me to configure the postUpgradeTasks I need in order to apply the assembly bindings after renovate have updated the dependencies.

The renovate.json config looks like this:

```
361 lines (359 sloc)  18.3 KB
 

{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabledManagers": [ "nuget"],
  "packageRules": [
    {
      "description": "Group all dependencies",
      "matchPaths": ["Source/**"],
      "groupName": "My dependencies",
      "postUpgradeTasks": {
        "commands": [
          "cd Source && curl -vLJO -H 'Accept:application/octet-stream' -H 'X-GitHub-Api-Version: 2022-11-28' https://api.github.com/repos/AdmincontrolAS/Add-BindingRedirect/releases/assets/102429039 -o Add-BindingRedirect && chmod 755 Add-BindingRedirect && ./Add-BindingRedirect && rm Add-BindingRedirect"
        ],
        "fileFilters": ["**/**"],
        "executionMode": "branch"
      }
    }
  ]
}
```

And in my global config.json I added:

```
{
    "allowPostUpgradeCommandTemplating": true,
    "allowedPostUpgradeCommands": ["Add-BindingRedirect"],
    "dependencyDashboardApproval":true
}
```

This basicly does the job, but  I am still testing this on larger projects.

**Comment by rarkins** [comment]
> the Add-Bindingredirect project from Andrew Harcourt

![image](https://user-images.githubusercontent.com/6311784/232973547-2000b81d-4220-4e4e-93c9-d7eda205640a.png)

@uglybugger
